controllers/core/openstackcontrolplane_controller.go

```diff
@@ -484,6 +486,13 @@ func (r *OpenStackControlPlaneReconciler) reconcileNormal(ctx context.Context, i
 		return ctrlResult, nil
 	}
 
+	ctrlResult, err = openstack.ReconcileLightSpeed(ctx, instance, version, helper)
```

Comment:
It isn't clear to me that the LightSpeed configuration is specifically related to our ControlPlane CRs. It seems you would have 1 instance of Lightspeed per cluster... not per namespace right? So I wonder if instead of having your integration point here it might make sense to do this in the OpenStack initialization resource which is 1 per cluster. The point of the existing initialization resource is to install operators but it could also create inject the openstack lightspeed configuration as well. It could be enabled/disabled via a parameter in the spec of that resource

Reply:
> It seems you would have 1 instance of Lightspeed per cluster... not per namespace right?

Yes, but it will not be the case in the future. 

We are currently limited by OpenShift Lightspeed, which is missing some functionality that does not allow us to do per-namespace configuration. However, the long-term goal is to allow different configurations of OpenStack Lightspeed for different OpenStack deployments on top of the same OpenShift cluster. 

Ultimately, OpenStack Lightspeed falls into a similar bucket as `Horizon` or `OpenStackClient`. Just a different interface to interact with OpenStack.

Reply:
I think it would be okay to have Lightspeed configured "alongside" the Controlplane. But I do not think it is a core service and thus should not be part of the Controlplane. I'm -2 on implementing the feature this way at this time.

Reply:
> But I do not think it is a core service and thus should not be part of the Controlplane.

Thanks for the response. In that case, I have some follow questions, please:

- What level of integration would be acceptable for you at this moment? Are we talking just the OpenStackLightspeed CR without integration in the OpenStackControlPlane CR?

- How does the OpenStack Lightpseed differ for you from Horizon or OpenStackClient? In the end, it is planned to be another interface for the deployed Openstack. I'm just trying to understand the reasoning.

The first one is probably the most important one because our team needs to find a way how to do this initial integration somehow.

Reply:
not because it is an AI component as much as it seems like it is a *new* component which could change greatly in a short period of time. We want the OpenStackControlplane to be as stable as we can make it.

If you need the configuration to be namespaced I think that is easily solved by simply having a CR in each namespace it is required. It doesn't mean we have to keep lightspeed in the same CR as the OpenStackControlplane right?

I think the suggestion here is somewhat minor. Keep the lightspeed functionality in the openstack-operator as a separate set API/controllers